test_utils/scripts/get_target_packages_kokoro.py

```diff
@@ -39,12 +40,37 @@ def get_base(environment):
         return 'HEAD~1'
 
 
-def get_changed_files(base):
+def get_changed_files_from_base(base):
     return subprocess.check_output([
         'git', 'diff', '--name-only', f'{base}..HEAD',
     ], stderr=subprocess.DEVNULL).decode('utf8').strip().split('\n')
 
 
+
+def parse_response_link(response):
+    header = response.headers.get("link")
+    if header is not None:
+        for link in header.split(","):
+            link = link.strip()
+            wrapped_url, rel = link.split(";")
+            if rel.strip() == 'rel="next"':
+                return wrapped_url[1:-1]  # strip <>
+
+_URL_TEMPLATE = (
+    'https://api.github.com/repos/googleapis/google-cloud-python/pulls/'
+    '{}/files'
+)
+
+
+def get_changed_files_from_pr(pr):
+    url = _URL_TEMPLATE.format(pr)
+    while url is not None:
+        response = requests.get(url)
+        for info in response.json():
+            yield info['filename']
+        url = parse_response_link(response)
```

Comment:
In Autorelease and Autosynth pagination is handled like this: Not sure if it would be possible to do it the same way here:

```
def list_org_repos(self, org: str, type: str = None) -> Sequence[dict]:
    url = f"{self.GITHUB_ROOT}/orgs/{org}/repos"
    
    while url:
        response = self.session.get(url, params={"type": type})
        response.raise_for_status()
        for item in response.json():
             yield item
   
        url = response.links.get("next", {}).get("url")

```

Reply:
Cool, I didn't know about the [`Response.links` feature in requests](https://2.python-requests.org/en/master/user/advanced/#link-headers).

Reply:
efa9411